rs/registry/canister/src/mutations/node_management/do_replace_operator.rs

```diff
@@ -0,0 +1,628 @@
+            upsert_node_operator_mutation(operator(2), provider(1), 10, "dc1"),
+            upsert_node_operator_mutation(operator(3), provider(1), 10, "dc1"),
+            upsert_node_mutation(node(1), operator(1)),
+            upsert_node_mutation(node(2), operator(3)),
+        ]);
+
+        registry
+            .do_replace_operator_with_caller(
+                payload(operator(1), operator(2), &[node(1), node(2)]),
+                caller(1),
+            )
+            .assert_err_contains(&format!(
+                "Node {} does not belong to node operator {}",
+                node(2),
+                operator(1)
+            ));
+    }
+
+    #[test]
+    fn insufficient_node_allowance() {
+        let mut registry = Registry::new();
+
+        registry.apply_mutations_for_test(vec![
+            upsert_dc_mutation("dc1"),
+            upsert_node_operator_mutation(operator(1), provider(1), 10, "dc1"),
+            upsert_node_operator_mutation(operator(2), provider(1), 2, "dc1"),
+            upsert_node_mutation(node(1), operator(1)),
+            upsert_node_mutation(node(2), operator(1)),
+            upsert_node_mutation(node(3), operator(1)),
+        ]);
+
+        registry
+            .do_replace_operator_with_caller(
+                payload(operator(1), operator(2), &[node(1), node(2), node(3)]),
+                caller(1),
+            )
+            .assert_err_contains("New operator cannot accept 3 nodes due to remaining allowance 2");
+    }
+
+    #[test]
+    fn update_all_records_correctly() {
+        let mut registry = invariant_compliant_registry(0);
+
+        let (atomic_request, mut node_ids) =
+            prepare_registry_with_nodes_and_node_operator_id(1, 3, operator(1));
+
+        registry.maybe_apply_mutation_internal(atomic_request.mutations);
+        let nodes_latest_version = registry.latest_version();
+
+        registry.maybe_apply_mutation_internal(vec![
+            upsert_dc_mutation("dc1"),
+            upsert_node_operator_mutation(operator(1), provider(1), 10, "dc1"),
+            upsert_node_operator_mutation(operator(2), provider(1), 10, "dc1"),
+        ]);
+
+        let first_node = node_ids.pop_first().map(|(key, _)| key).unwrap();
+        let second_node = node_ids.pop_first().map(|(key, _)| key).unwrap();
+        let third_node = node_ids.pop_first().map(|(key, _)| key).unwrap();
+
+        let version_before = registry.latest_version();
+        registry
+            .do_replace_operator_with_caller(
+                payload(operator(1), operator(2), &[first_node, second_node]),
+                caller(1),
+            )
+            .assert_ok();
+
+        assert!(
+            registry.latest_version() == version_before + 1,
+            "Expected registry version to increase. Before execution: {}, after execution: {}",
+            version_before,
+            registry.latest_version()
+        );
```

Comment:
No action required.

I would push all these version checks down, because these changes are implicit/secondary. Ofc, they are also critical, but still, secondary.